components/packages/index.rst

```diff
@@ -0,0 +1,259 @@
+
+.. code-block:: yaml
+
+    # In config.yaml
+    substitutions:
+      node_name: mydevice
+      device_verbose_name: "My Device"
+
+    packages:
+      wifi: !include common/wifi.yaml
+      device_base: !include common/device_base.yaml
+
+    api:
+      services:
+        - service: start_laundry
+          then:
+            - switch.turn_on: relay
+            - delay: 3h
+            - switch.turn_off: relay
+
+    sensor:
+      - platform: mhz19
+        co2:
+          name: "CO2"
+        temperature:
+          name: "Temperature"
+        update_interval: 60s
+        automatic_baseline_calibration: false
+
+.. code-block:: yaml
+
+    # In wifi.yaml
+    wifi:
+      ssid: !secret wifi_ssid
+      password: !secret wifi_password
+      domain: .yourdomain.lan
+      fast_connect: true
+
+.. code-block:: yaml
+
+    # In device_base.yaml
+    esphome:
+      name: ${node_name}
+      platform: ESP32
+      board: wemos_d1_mini32
+      build_path: ./build/${node_name}
+
+    # I²C Bus
+    i2c:
+      sda: GPIOXX
+      scl: GPIOXX
+      scan: true
+      frequency: 100kHz
+
+    # Enable logging
+    logger:
+      level: ${log_level}
+
+    api:
+      encryption:
+        key: !secret api_encryption_key
+      reboot_timeout: 1h
+
+    sensor:
+      - <<: !include common/sensor/uptime.config.yaml
+      - <<: !include common/sensor/wifi_signal.config.yaml
+    binary_sensor:
+      - <<: !include common/binary_sensor/connection_status.config.yaml
+
+    switch:
+      - <<: !include common/switch/restart_switch.config.yaml
+
+.. _config-git_packages:
+
+Remote/Git Packages
+-------------------
+
+Packages can also be loaded from a Git repository by utilizing the correct configuration syntax.
+:doc:`/components/substitutions/index` can be used inside the remote packages which allows users to override
+them locally with their own substitution value.
+
+.. note::
+
+    Remote packages cannot have ``secret`` lookups in them. They should instead make use of substitutions with an
+    optional default in the packaged YAML, which the local device YAML can set using values from the local secrets.
+
+.. code-block:: yaml
+
+    packages:
+      # Git repo examples
+      remote_package:
+        url: https://github.com/esphome/non-existant-repo
+        ref: main # optional
+        files: [file1.yml, file2.yml]
+        refresh: 1d # optional
+
+      # A single file can be expressed using `file` or `files` as a string
+      remote_package_two:
+        url: https://github.com/esphome/non-existant-repo
+        file: file1.yml # cannot be combined with `files`
+        # files: file1.yml
+
+      # shorthand form github://username/repository/[folder/]file-path.yml[@branch-or-tag]
+      remote_package_three: github://esphome/non-existant-repo/file1.yml@main
+
+Packages as Templates
+---------------------
+
+Since packages are incorporated using the ``!include`` system, variables can be provided to them. This means that
```

Comment:
When i checked this is the first time ``!include`` is mentioned, would it be a good idea to explains what the differences are (when there are off-coerce).